src/main/java/com/api/controller/RocketController.java

```diff
@@ -26,7 +36,28 @@ public List<RocketDto> getAllRockets(
             @RequestParam(value = "limit", required = false) Integer limitParam,
             @RequestParam(value = "offset", required = false) Integer offsetParam
     ) {
-        return rocketPresenter.getAllRockets(idParam, limitParam, offsetParam);
+
+        List<RocketDto> rocketDtoList = rocketPresenter.getAllRockets(idParam, limitParam, offsetParam);
+        List<SxRocket> sxRocketList = rocketService.getRockets(rocketDtoList);
+        rocketPersistenceService.saveRockets(sxRocketList);
```

Comment:
this should be in the controller, Controller is reponsible only for the UI layer and UI updates/checks